### PR TITLE
Add default window_strides and padding to lax.reduce_window.

### PR DESCRIPTION
**Pull request by carlosgmartin**
Add useful defaults to [`lax.reduce_window`](https://docs.jax.dev/en/latest/_autosummary/jax.lax.reduce_window.html):
- `window_strides=None`: sets it to `(1,) * len(window_dimensions)`, exactly like the default value `None` does for `base_dilation` and `window_dilation`.
- `padding="VALID"`: no padding.

### CODE REVIEW

File: jax/_src/lax/windowed_reductions.py [hunk 1]
```diff
@@ -48,8 +48,8 @@ def _reduce_window(
     init_value,
     computation,
     window_dimensions: core.Shape,
-    window_strides: Sequence[int],
-    padding: str | Sequence[tuple[int, int]],
+    window_strides: Sequence[int] | None = None,
+    padding: str | Sequence[tuple[int, int]] = "VALID",
```
Comment (jakevdp) [inline jax/_src/lax/windowed_reductions.py]: Probably better for this API to not have defaults, as it's only used internally, and not passing one of the arguments would likely indicate a coding error.

Comment (carlosgmartin) [reply jax/_src/lax/windowed_reductions.py]: `reduce_window` is used externally by various libraries:
- [`flax.linen.pool`](https://flax-linen.readthedocs.io/en/latest/api_reference/flax.linen/layers.html#flax.linen.pool): [line](https://github.com/google/flax/blob/8c22f29573edf90ed41283a4330a919742bb40be/flax/linen/pooling.py#L73)
- [`haiku.avg_pool`](https://dm-haiku.readthedocs.io/en/latest/api.html#haiku.avg_pool): [line](https://github.com/google-deepmind/dm-haiku/blob/8abeef4edf512d88546ec95fb308c597d2cd8996/haiku/_src/pool.py#L135)
- [`haiku.max_pool`](https://dm-haiku.readthedocs.io/en/latest/api.html#haiku.max_pool): [line](https://github.com/google-deepmind/dm-haiku/blob/8abeef4edf512d88546ec95fb308c597d2cd8996/haiku/_src/pool.py#L101)
- [`equinox.nn.Pool.__call__`](https://docs.kidger.site/equinox/api/nn/pool/#equinox.nn.Pool.__call__): [line](https://github.com/patrick-kidger/equinox/blob/590c88ba7a836284e78b5afcc26ef405941abf54/equinox/nn/_pool.py#L143)
- [`jaxnet`](https://github.com/JuliusKunze/jaxnet): [line](https://github.com/juliuskunze/jaxnet/blob/3df8a788a90003b48c3a136e0b68d0e8f1fdd679/jaxnet/modules.py#L127)
- [`zephyr`](https://github.com/mzguntalan/zephyr): [line](https://github.com/mzguntalan/zephyr/blob/21ae066128cb4b6122c1b37302f0b9c2578410a1/zephyr/_nets/pool.py#L32)

`flax` and `jaxnet` share the following logic:
```python3
if strides is None:
  strides = (1,) * len(window_shape)
```

This makes `strides` symmetrical with `base_dilation` and `window_dilation` in terms of how `None` is handled: Just create a tuple of 1s of the same length as `window_shape`. This is a very sensible default, I think.

Comment (carlosgmartin) [reply jax/_src/lax/windowed_reductions.py]: Oh, I think you specifically meant the [private function](https://github.com/jax-ml/jax/blob/6f8456e9fbf776dc85f5d7cf9b0bade9c658a76c/jax/_src/lax/windowed_reductions.py#L46), not the [public wrapper](https://github.com/jax-ml/jax/blob/6f8456e9fbf776dc85f5d7cf9b0bade9c658a76c/jax/_src/lax/windowed_reductions.py#L113). Is that right?

Out of curiosity, though, is there any reason to break the equivalence between the two? In fact, is there any reason for the wrapper at all? They seem to have identical signatures.

Comment (jakevdp) [reply jax/_src/lax/windowed_reductions.py]: Yes, I'm talking about the private `_reduce_window` function. I think the change to the public `reduce_window` function is fine.

The main reason I'd break equivalence between them is because `_reduce_window` is only used in contexts where arguments are forwarded from another function call. Leaving out an argument in this context is problematic, and so default values allow for silent bugs that would otherwise fail loudly.

Comment (carlosgmartin) [reply jax/_src/lax/windowed_reductions.py]: I see. To clarify, would you like me to move the default values (and `None` handling) for the other arguments (`base_dilation` and `window_dilation`) from `_reduce_window` to `reduce_window`, as well?

Comment (jakevdp) [reply jax/_src/lax/windowed_reductions.py]: I wouldn't move the `None` handling – it's fine for that to be part of the private implementation. This PR is about adding default values to the public APIs – you can probably just leave it at that.

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
Thanks again for this!
